README.md

```diff
@@ -1,19 +1,83 @@
+
+## Disambiguation: connectors-ruby and connectors-python
+
+This repository contains connector clients written in Ruby. However, some connectors will also be implemented in Python. The Python connectors are located in a separate repository, [elastic/connectors-python](https://github.com/elastic/connectors-python).
+
+Before getting started, review important information about this feature:
+
+- [Known issues and limitations](#known-issues-and-limitations)
+- [Getting help and providing feedback](#getting-help-and-providing-feedback)
+
+Build, deploy, and operate your connector:
+
+- [Building a connector](#building-a-connector)
+- [Moving to production](#moving-to-production)
+- [Publishing a connector (optional)](#publishing-a-connector-optional)
+
+Reference:
+
+- [Connector protocol reference](#connector-protocol-reference)
+
+## Known issues and limitations
+
+[//]: # (If the only "limitation" is technical preview, we could drop this section.)
+- The connector clients implemented in this repository are [mongodb](/lib/connectors/mongodb), [gitlab](/lib/connectors/gitlab) and [stub connector](lib/connectors/stub_connector). None of those are production ready clients - rather, they are functional examples, which you can use as a base for your own work.
+- The `stub_connector` is the most basic example that doesn't have any integration with the third-party data source - it just returns some stub data to ingest.
+- The `mongodb` connector is a MongoDB-based connector. It is lacking some features that one would definitely want in production - for example it does not implement any kind of authentication and it only works with a single collection, but it is still a good example of how to use the connector framework for handling the data from this kind of a document storage.
+- The `gitlab` connector is the most involved out of the three but is is also missing a lot of features - for one, it only grabs the projects and leaves aside all the other types of content, like files, folders, issues etc. In the current simplified version, it also doesn't synchronize document permissions. But it implements the authentication via an API token, and it has the most developed class structure, which is easily extendable.
+
+## Getting help and providing feedback
+
+May want to duplicate the content I put in the Elastic doc: https://github.com/elastic/enterprise-search-pubs/pull/2631
+
+## Building a connector
+
+This section has its own structure, so mini TOC to provide an overview:
+
+- [Implementing the connector protocol](#implementing-the-connector-protocol)
+- [Using the connector framework](#using-the-connector-framework)
+- [Operating the connector service](#operating-the-connector-service)
+- [Operating the connector](#operating-the-connector)
+- [Testing the integration](#testing-the-integration)
+
+### Implementing the connector protocol
```

Comment:
Users would benefit from more context. Fundamentally, to create an Elastic connector, you need to implement this protocol. At a high level, you need to create an application (the connector) that can read and write to a specific document in Elasticsearch that represents the connector. And you need to log sync jobs to an additional index.

Reply:
Done.